src/NUnitFramework/framework/Interfaces/AssertionResult.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NUnit.Framework.Interfaces
+{
+    /// <summary>
+    /// The AssertionResult class represents the result of a single assertion.
+    /// </summary>
+    public class AssertionResult
+    {
+        /// <summary>
+        /// Construct an AssertionResult
+        /// </summary>
+        public AssertionResult(AssertionStatus status, string message, string stackTrace)
+        {
+            Status = status;
+            Message = message;
+            StackTrace = stackTrace;
+        }
+
+        /// <summary> The pass/fail status of the assertion</summary>
+        public AssertionStatus Status { get; private set; }
+
+        /// <summary>The message produced by the assertion, or null</summary>
+        public string Message { get; private set; }
+
+        /// <summary>The stacktrace associated with the assertion, or null</summary>
+        public string StackTrace { get; private set; }
+
+        /// <summary>
+        /// ToString Override
+        /// </summary>
+        public override string ToString()
```

Comment:
Purely meant for debugging. I generally don't add stuff to the SUT for test perposes, but I make an exception for ToString(), Equals and GetHashcode because they are so commonly needed and relatively easy to implement in most cases.